### PR TITLE
🪴 Rancher credentials lookup by namespace name

### LINKED ISSUE #386: Enabling digital ocean provider fails
### What steps did you take and what happened?

When trying to enable the digital ocean CAPI provider referencing a Rancher Cloud Credential we are seeing this error:

```
E0216 14:03:50.520691       1 secret_mapper_sync.go:279]  "msg"="failed to map credential keys" "error"="key not  │
│ found: digitaloceancredentialConfig-accessToken" "CAPIProvider"={"name":"digitalocean","namespace":"capdo-system" │
│ } "controller"="capiprovider" "controllerGroup"="turtles-capi.cattle.io" "controllerKind"="CAPIProvider" "name"=" │
│ digitalocean" "namespace"="capdo-system" "reconcileID"="36c806a1-6231-4121-9cb5-bdedc6315c6b"                     │
│ E0216 14:03:50.520739       1 controller.go:329]  "msg"="Reconciler error" "error"="unable to locate rancher secr │
│ et with name cattle-global-data:cc-7zfpj for provider digitalocean" "CAPIProvider"={"name":"digitalocean","namesp │
│ ace":"capdo-system"} "controller"="capiprovider" "controllerGroup"="turtles-capi.cattle.io" "controllerKind"="CAP │
│ IProvider" "name"="digitalocean" "namespace"="capdo-system" "reconcileID"="36c806a1-6231-4121-9cb5-bdedc6315c6b"
```

### What did you expect to happen?

The expectation is the provider is installed and the credentials from the Rancher Cloud Credential are used when enabling the DO provider.

### How to reproduce it?

1. Setup Rancher Manager (tested with v2.8.2)
2. Install Rancher Turtles (tested with v0.5.0)
3. Create a DO cloud credential in Rancher manager
4. Create a namespace:

```bash
kubectl create ns capdo-system
```

6. Apply the following yaml to enable the DO CAPI provider

> **You will need to change the name of the cloud credential from cc-7zfpj

```yaml
apiVersion: turtles-capi.cattle.io/v1alpha1
kind: CAPIProvider
metadata:
  name: digitalocean
  namespace: capdo-system
spec:
  # configSecret:
  #   name: digitalocean-credentials-wnmr7
  credentials:
    rancherCloudCredential: cattle-global-data:cc-7zfpj
  features:
    clusterResourceSet: true
    clusterTopology: true
    machinePool: true
  name: digitalocean
  type: infrastructure
  variables: {}
```

### Rancher Turtles version

v0.5.0

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind bug

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:
Implementing a separate field for a fully qualified `namespace:name` reference to the rancher cloud credentials secret, as an alternative to the name annotation.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #386 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: api/v1alpha1/capiprovider_types.go [hunk 1]
```diff
@@ -79,13 +79,18 @@ type Features struct {
 // Credentials defines the external credentials information for the provider.
 // +kubebuilder:validation:MaxProperties=1
 // +kubebuilder:validation:MinProperties=1
+// +kubebuilder:validation:XValidation:message="rancherCloudCredentialNamespaceName should be in the namespace:name format.",rule="!has(self.rancherCloudCredentialNamespaceName) || self.rancherCloudCredentialNamespaceName.matches('^.+:.+$')"
```
Comment (richardcase) [inline api/v1alpha1/capiprovider_types.go:82]: Nice :)

File: api/v1alpha1/capiprovider_types.go [hunk 2]
```diff
@@ -79,13 +79,18 @@ type Features struct {
...
 // +structType=atomic
 //
 //nolint:godot
+//nolint:lll
 type Credentials struct {
 	// RancherCloudCredential is the Rancher Cloud Credential name
 	RancherCloudCredential string `json:"rancherCloudCredential,omitempty"`
 
+	// RancherCloudCredentialNamespaceName is the Rancher Cloud Credential namespace:name reference
+	RancherCloudCredentialNamespaceName string `json:"rancherCloudCredentialNamespaceName,omitempty"`
```
Comment (richardcase) [inline api/v1alpha1/capiprovider_types.go:92]: Maybe in the future with an api bump we can move both these under a `Rancher` struct

**Review by richardcase — APPROVED** [review]
Looks good to me.

File: api/v1alpha1/capiprovider_types.go (hunk 2 above)
Comment (Danil-Grigorev) [reply api/v1alpha1/capiprovider_types.go:92]: Yeah, makes sense as a separate the field entirely.

**Review by salasberryfin — APPROVED** [review]

**Pull request merged**